packages/app-page-builder/src/editor/plugins/elements/image/index.tsx

```diff
@@ -66,6 +66,8 @@ export default (args: PbEditorElementPluginArgs = {}): Plugin[] => {
                     ? args.settings(defaultSettings)
                     : defaultSettings,
             target: ["cell", "block"],
+            allowDynamicSource: true,
```

Comment:
Let's restructure this a bit. When you see a plugin with all of its fields, like `name`, `type`, `toolbar`, `settings`, and then all of the sudden `allowedFields`, it's not immediately obvious what is the context of this property. 

Since it's related to dynamic pages, instead of having:

```ts
allowDynamicSource: boolean;
allowedFields: string[]
```,

...let's maybe have just:

```ts
dynamicDataSource: {
   enabled: boolean
   allowedFields: string[]
}
```

Wdyt?